### DISCUSSION TITLE
How do i set up my mqtt client and server with domain-validated certificate?

### DISCUSSION THREAD

**Discussion by wilari932** [dotnet/MQTTnet#1409] [Q&A]
Hi im writing a a client and a sever that are going to communicate throw TLS with a domain-validated certificate.
In the wiki i only found exempels how to set upp with self signed certificate, i dont want to compile a pem file with my clients do you have any solution to this ?

**Comment by SeppPenner** [comment]
A domain-validated certificate is a "regular" certificate that is issued for a domain, right? Like a certificate issued for www.example.com?

With a "regular", not self-signed certificate, you don't need to add the certificate, the standard TLS connections are validated by .Net already.

**Reply by wilari932** [reply]
do i need to setup like this for the client:  ?
![image](https://user-images.githubusercontent.com/33053518/167947103-5ea250a7-a64a-4b23-8f69-200eafa4769c.png)
for the server
![image](https://user-images.githubusercontent.com/33053518/167947485-3033d9f7-b187-452f-9b9c-53e90ebc708d.png)
in order to get tls working, with a "regular" cert?

**Reply by SeppPenner** [reply]
Yes, like that. Check out https://github.com/dotnet/MQTTnet/wiki/Client#secure-tcp-connection as well. If you want to run the server on port 1884 (Second screenshot), you would need to use `.WithTcpServer("broker.example.com", 1884).WithTls()` to specify the port.

**Reply by wilari932** [reply]
Thansk but when i run the code for the server second image it throws an exception says that i need to specify a cert :) ?

**Reply by SeppPenner** [reply]
Can you test if your setup works with a public broker like `broker.emqx.io` with port 8883 following https://www.emqx.com/en/mqtt/public-mqtt5-broker?

Another thing that might be a problem: Is your server certificate signed by a well-known CA (certificate authority like e.g. LetsEncrypt, Commodo, ...) or not? If not, you might need to supply the CA certificate for the clients...
